### PR TITLE
Fixes #23085: rudder-compliance python script does not support techniques with blocks when some of the methods inside report an error

### PR DESCRIPTION
**Pull request by Fdall**
https://issues.rudder.io/issues/23085

### CODE REVIEW

**Comment by Fdall** [comment]
PR updated with a new commit

**Comment by Fdall** [comment]
PR rebased

File: scripts/rudder-compliance/rudder-compliance [hunk 1]
```diff
@@ -70,24 +70,51 @@ def get_non_compliance_rule(rule_compliance, non_compliances):
...
-                  if any (k in reports["status"] for k in non_compliances_keys):
-                    non_compliances.append([rule_name, directive_name, component_name, node_name, values["value"], reports["status"], reports.get("message", "")])
+  # Look if any directive is non compliant
+  if any (k in directive_compliance['complianceDetails'] for k in non_compliances_keys):
+    for components_compliance in directive_compliance["components"]:
+      # Look for the faulty component
+      if any (k in components_compliance['complianceDetails'] for k in non_compliances_keys):
+        # Use recursion to get the per node compliance as it can be nested in blocks
+        for i in walk_through_blocks(components_compliance):
+          non_compliances.append(
```
Comment (ncharles) [inline scripts/rudder-compliance/rudder-compliance:110]: this syntax adds None in the list

Comment (ncharles) [reply scripts/rudder-compliance/rudder-compliance:110]:
```suggestion
          entry = [rule_name, directive_compliance["name"], components_compliance["name"]]
          entry.extend(i)
          non_compliances.append(entry)

```

**Review by ncharles — APPROVED** [review]

File: scripts/rudder-compliance/rudder-compliance [hunk 2]
```diff
@@ -70,24 +70,54 @@ def get_non_compliance_rule(rule_compliance, non_compliances):
...
+        for i in walk_through_blocks(components_compliance):
+          entry = [rule_name, directive_compliance["name"], components_compliance["name"]]
+          entry.extend(i)
+          non_compliances.append(entry)
+
+            [
+              rule_name,
+              directive_compliance["name"],
+              components_compliance["name"]
+            ].extend(i)
```
Comment (ncharles) [inline scripts/rudder-compliance/rudder-compliance:118]: wait, what is this ? it should not have stayed there

**Pull request merged**